### ISSUE TITLE
[Training] Inquiry About Data Augmentation Method in Your Dataset Class

### ISSUE THREAD

**Issue by 15253a** [KAIST-AILab/SyncVSR#27]
Dear Author,

I hope I don't bother you too much.

I have a question about data augmentation in your datasets class. The code snippet is as follows
        
        _filename = self.filenames[idx]
        data = torch.load(filename)

        # load from dictionary
        video = data["video"]
        video_size = len(video)
        audio = data["audio"]
        text = data["text"]
        text_data = text.split("\n")

        if video_size > self.cut: # pretrain or vox2
            # shard the text
            margin = 0.5 if "LRS3" in filename else 0.2

            # get random starting point and crop video and audio
            random_start = random.randint(0, video_size - self.cut)
            sample_length = np.random.choice(len(self.length_distribution), p=self.probability_distribution)
            video = video[random_start:random_start+sample_length]
            audio = audio[random_start*self.audio_multiple:(random_start+sample_length)*self.audio_multiple]

            # parse text for pretrain files with timestamps
            words = [text_data[i].split(" ") for i in range(4, len(text_data)-1)]
            
            # cut and align video, text and audio
            time_start = random_start / 25
            time_end = time_start + (sample_length / 25)
            
            list_word = []
            for word in words:
                if (time_start - margin) <= float(word[1]) and float(word[2]) < (time_end + margin):
                    list_word.append(word[0])
            
            text_string = " ".join(list_word)
    
        else: # main or short vox2
            text_string = text_data[0][5:].strip()_

 Could you please clarify whether the method of randomly trimming the data, as demonstrated above, has a significant impact on the model's performance? I ask this because the label texts in my own dataset do not include timestamp information, and I am concerned about potential discrepancies this might introduce.
 I would greatly appreciate any insights or recommendations you could provide on this matter.Thank you very much for your time and assistance.

**Comment by snoop2head** [comment]
Dear @15253a ,

Thank you for taking an interest in our work and for your question regarding data augmentation. Datasets like Vox2 and LRS3 often include sequences that are significantly longer than the typical training window we can handle. Because managing such lengthy sequences can be computationally prohibitive, we did not conduct a direct ablation study comparing the performance impact of trimming versus using full sequences.

For Vox2, we generated the timestamp information by transcribing with Whisper (refer to our script at [LRS/video/preprocess/transcribe_whisper.py](https://github.com/KAIST-AILab/SyncVSR/blob/main/LRS/video/preprocess/transcribe_whisper.py). You may generate timestamps for your dataset referring to the code. However, if you have sufficient computational resources to handle full-length sequences for both video frames and text tokens, you might consider omitting the trimming step altogether. 

I hope this clarifies our approach. If you have any further questions or concerns, please feel free to let us know!

**Comment by 15253a** [comment]
thanks
